### PR TITLE
Service updates with SharedPermissions and UserTypes

### PR DESCRIPTION
**Pull request by zcox10**
## Description
Updates to pharmaid to include all necessary enums (`UserType`, `SharePermissionType`, and `ShareRequestStatus`).  Added validation logic in `SharedPermissionValidator` to ensure the proper users / actions are taking place when making shared requests. The validation checks are utilized also in PrescriptionService when a user attempts to VIEW/EDIT another user's prescriptions. 

## Checklist

- [X] Tests pass
- [X] No checkstyle errors
- [X] No PMD errors
- [X] No compilation/build errors

### CODE REVIEW

File: src/main/java/com/bytecoders/pharmaid/util/ServiceUtils.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  /**
+   * Finds an entityId or throws ResponseStatusException if not found.
+   *
+   * @param <T>        Entity type
+   * @param id         The ID of the entity
+   * @param entityName The name of the entity (eg "user", "medication")
+   * @param repository The repository for the entity
+   * @return The entity if found
+   */
+  public <T> T findEntityById(
```
Comment (AlexLoyko) [inline src/main/java/com/bytecoders/pharmaid/util/ServiceUtils.java:25]: nice abstraction

File: src/main/java/com/bytecoders/pharmaid/service/SharedPermissionValidator.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+  /**
+   * Processes all validations to ensure proper setup for a share request.
+   *
+   * @param owner          the {@link User} who owns the health records
+   * @param requester      the {@link User} requesting access to the health records
+   * @param permissionType the {@link SharePermissionType} the requester is seeking to obtain
+   */
+  public void validateCreateShareRequestSetup(
+      User owner, User requester, SharePermissionType permissionType) {
+    validateLoggedInUser(jwtUtils.getLoggedInUserId(), requester.getId());
```
[the remark this answers is not in the document]
Comment (zcox10) [reply src/main/java/com/bytecoders/pharmaid/service/SharedPermissionValidator.java:223]: It throws an exception if it already exists with [validateExistingPermission](https://github.com/NavinColumbia/COMS4156-TeamProject-ByteCoders/pull/32/files#diff-52d3d4fcd54f59f3125ac98f7e74194348e1ed7781ab6be2857ac1cd53e3118fR174).  Is that ok or do you think we should return the request instead?

[the remark this answers is not in the document]
Comment (AlexLoyko) [reply src/main/java/com/bytecoders/pharmaid/service/SharedPermissionValidator.java:223]: ah i missed that

**Review by AlexLoyko — APPROVED** [review]

File: src/main/java/com/bytecoders/pharmaid/GlobalExceptionHandler.java [hunk 3]
```diff
@@ -24,8 +29,48 @@ public class GlobalExceptionHandler {
...
+  /**
+   * Handler for invalid enum values or other illegal arguments.
+   *
+   * @param ex IllegalArgumentException
+   * @return a ResponseEntity indicating BAD_REQUEST with a detailed message
+   */
+  @ExceptionHandler(IllegalArgumentException.class)
+  public ResponseEntity<Map<String, String>> handleIllegalArgumentException(
+      IllegalArgumentException ex) {
+    System.out.println("Caught IllegalArgumentException: " + ex.getMessage());
```
[the remark this answers is not in the document]
Comment (zcox10) [reply src/main/java/com/bytecoders/pharmaid/GlobalExceptionHandler.java]: updated to log.error()

**Pull request merged**